src/cmd/pay.rs

```diff
@@ -259,15 +276,62 @@ pub enum TokenInput {
     Hst,
 }
 
-// By default, helium-api-rs serializes and deserializes tokens as u64s (ie: bones).
-// This overrides the default serializer when parsing JSON.
-// Structopt deserializes using "from_str", which also expects Decimal representation
-fn token_decimal_deserializer<'de, D>(deserializer: D) -> std::result::Result<Token, D::Error>
-where
-    D: serde::Deserializer<'de>,
-{
-    let decimal: rust_decimal::Decimal = Deserialize::deserialize(deserializer)?;
-    Ok(Token::new(decimal))
+#[derive(Debug, PartialEq)]
+// An amount can be either the string "max" or a number, the deserializer handles both cases.
+// token_amount() will return 0 (when Amount::Max) or a valid token amount. Use this instead
+// of Token::Dec directly.
+//
+// We use decimal for all numbers because the helium-api-rs serializes and deserializes tokens as
+// u64s (ie: bones).
+enum Amount {
+    Dec(Decimal),
+    Max,
+}
+
+impl Amount {
+    pub fn token_amount(&self) -> Token {
+        let d: Decimal = match self {
+            Amount::Dec(raw) => *raw,
+            _ => Decimal::from(0),
+        };
+        Token::new(d.round_dp(8))
```

Comment:
I had to use `round_dp(8)` here because the unit tests failed on the `1.6` input. I don't understand why it didn't fail before. I could use some advice here because this is crappy.